README
NAME
    Dicom::File::Detect - Detect DICOM file through magic string.

SYNOPSIS
     use Dicom::File::Detect qw(dicom_detect_file);

     my $dcm_flag = dicom_detect_file($file);

DESCRIPTION
    This Perl module detect DICOM file through magic string. DICOM (Digital
    Imaging and Communications in Medicine) is a standard for handling,
    storing, printing, and transmitting information in medical imaging. See
    DICOM on Wikipedia <https://en.wikipedia.org/wiki/DICOM>.

SUBROUTINES
  "dicom_detect_file"
     my $dcm_flag = dicom_detect_file($file);

    Detect DICOM file.

    Returns 1/0.

ERRORS
     dicom_detect_file():
             Cannot close file '%s'.
             Cannot open file '%s'.

EXAMPLE1
     use strict;
     use warnings;

     use Dicom::File::Detect qw(dicom_detect_file);
     use File::Temp qw(tempfile);
     use IO::Barf qw(barf);
     use MIME::Base64;

     # Data in base64.
     my $data = <<'END';
     AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
     AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
     AAAAAAAAAAAAAAAAAABESUNNCg==
     END

     # Temp file.
     my (undef, $temp_file) = tempfile();

     # Save data to file.
     barf($temp_file, decode_base64($data));

     # Check file.
     my $dcm_flag = dicom_detect_file($temp_file);

     # Print out.
     if ($dcm_flag) {
             print "File '$temp_file' is DICOM file.\n";
     } else {
             print "File '$temp_file' isn't DICOM file.\n";
     }

     # Output like:
     # File '%s' is DICOM file.

EXAMPLE2
     use strict;
     use warnings;

     use Dicom::File::Detect qw(dicom_detect_file);

     # Arguments.
     if (@ARGV < 1) {
             print STDERR "Usage: $0 file\n";
             exit 1;
     }
     my $file = $ARGV[0];

     # Check file.
     my $dcm_flag = dicom_detect_file($file);

     # Print out.
     if ($dcm_flag) {
             print "File '$file' is DICOM file.\n";
     } else {
             print "File '$file' isn't DICOM file.\n";
     }

     # Output:
     # Usage: dicom-detect-file file

DEPENDENCIES
    Error::Pure, Exporter, Readonly.

SEE ALSO
    File::Find::Rule::Dicom
        Common rules for searching for DICOM things.

    Task::Dicom
        Install the Dicom modules.

REPOSITORY
    <https://github.com/michal-josef-spacek/Dicom-File-Detect>

AUTHOR
    Michal Josef Špaček <mailto:skim@cpan.org>

    <http://skim.cz>

LICENSE AND COPYRIGHT
    © 2014-2021 Michal Josef Špaček

    BSD 2-Clause License

VERSION
    0.06



Changes
0.06

0.05 2021-02-07T20:28:21+01:00
 - Fix bugtracker.
 - Update copyright years.

0.04 2020-04-05T16:59:39+02:00
 - Fix Makefile.PL fails with no '.' in @INC.
 - Improve LICENSE AND COPYRIGHT section in doc.
 - Improve SYNOPSIS section in doc.
 - Rewrite doc to using POD =head2 for methods.
 - Update Module::Install to 1.19 version.
 - Update author github username.
 - Update author name.
 - Update copyright years.

0.03 2015-10-13T23:32:55+02:00
 - Fix test.
 - Fix test dependencies.
 - Improve LICENSE AND COPYRIGHT section in doc.
 - Improve SEE ALSO section in doc.
 - Move author tests to xt/ directory.
 - Remove diacritics from author name in Makefile.PL.
 - Update Module::Install to 1.16 version.
 - Update copyright years.

0.02 2014-08-07T09:25:47+02:00
 - Add DESCRIPTION section to doc.
 - Add File::Find::Rule::Dicom to SEE ALSO section in doc.
 - Add minimum Perl version.

0.01 2014-08-05T16:59:33+02:00
 - First version.


Detect.pm
package Dicom::File::Detect;

use base qw(Exporter);
use strict;
use warnings;

use Error::Pure qw(err);
use Readonly;

# Constants.
Readonly::Array our @EXPORT => qw(dicom_detect_file);
Readonly::Scalar our $DCM_MAGIC => qw{DICM};

our $VERSION = 0.06;

# Detect DICOM file.
sub dicom_detect_file {
	my $file = shift;

	my $dcm_flag = 0;
	open my $fh, '<', $file or err "Cannot open file '$file'.";
	my $seek = seek $fh, 128, 0;
	my $magic;
	my $read = read $fh, $magic, 4;
	close $fh or err "Cannot close file '$file'.";

	if ($magic eq $DCM_MAGIC) {
		return 1;
	} else {
		return 0;
	}
}

1;

__END__

=pod

=encoding utf8

=head1 NAME

Dicom::File::Detect - Detect DICOM file through magic string.

=head1 SYNOPSIS

 use Dicom::File::Detect qw(dicom_detect_file);

 my $dcm_flag = dicom_detect_file($file);

=head1 DESCRIPTION

This Perl module detect DICOM file through magic string.
DICOM (Digital Imaging and Communications in Medicine) is a standard for
handling, storing, printing, and transmitting information in medical imaging.
See L<DICOM on Wikipedia|https://en.wikipedia.org/wiki/DICOM>.

=head1 SUBROUTINES

=head2 C<dicom_detect_file>

 my $dcm_flag = dicom_detect_file($file);

Detect DICOM file.

Returns 1/0.

=head1 ERRORS

 dicom_detect_file():
         Cannot close file '%s'.
         Cannot open file '%s'.

=head1 EXAMPLE1

 use strict;
 use warnings;

 use Dicom::File::Detect qw(dicom_detect_file);
 use File::Temp qw(tempfile);
 use IO::Barf qw(barf);
 use MIME::Base64;

 # Data in base64.
 my $data = <<'END';
 AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
 AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
 AAAAAAAAAAAAAAAAAABESUNNCg==
 END

 # Temp file.
 my (undef, $temp_file) = tempfile();

 # Save data to file.
 barf($temp_file, decode_base64($data));

 # Check file.
 my $dcm_flag = dicom_detect_file($temp_file);

 # Print out.
 if ($dcm_flag) {
         print "File '$temp_file' is DICOM file.\n";
 } else {
         print "File '$temp_file' isn't DICOM file.\n";
 }

 # Output like:
 # File '%s' is DICOM file.

=head1 EXAMPLE2

 use strict;
 use warnings;

 use Dicom::File::Detect qw(dicom_detect_file);

 # Arguments.
 if (@ARGV < 1) {
         print STDERR "Usage: $0 file\n";
         exit 1;
 }
 my $file = $ARGV[0];

 # Check file.
 my $dcm_flag = dicom_detect_file($file);

 # Print out.
 if ($dcm_flag) {
         print "File '$file' is DICOM file.\n";
 } else {
         print "File '$file' isn't DICOM file.\n";
 }

 # Output:
 # Usage: dicom-detect-file file

=head1 DEPENDENCIES

L<Error::Pure>,
L<Exporter>,
L<Readonly>.

=head1 SEE ALSO

=over

=item L<File::Find::Rule::Dicom>

Common rules for searching for DICOM things.

=item L<Task::Dicom>

Install the Dicom modules.

=back

=head1 REPOSITORY

L<https://github.com/michal-josef-spacek/Dicom-File-Detect>

=head1 AUTHOR

Michal Josef Špaček L<mailto:skim@cpan.org>

L<http://skim.cz>

=head1 LICENSE AND COPYRIGHT

© 2014-2021 Michal Josef Špaček

BSD 2-Clause License

=head1 VERSION

0.06

=cut


Makefile.PL
use lib '.';
use strict;
use warnings;

use inc::Module::Install;

# Definition.
abstract 'Detect DICOM file through magic string.';
author 'Michal Josef Spacek <skim@cpan.org>';
author_requires 'English' => 0;
author_requires 'File::Object' => 0.05;
author_requires 'Test::More' => 0;
author_requires 'Test::NoWarnings' => 0;
author_requires 'Test::Pod' => 0;
author_requires 'Test::Pod::Coverage' => 0;
license 'bsd';
name 'Dicom-File-Detect';
requires 'Readonly' => 0;
requires 'perl' => '5.8.0';
readme_from 'Detect.pm';
recursive_author_tests('xt');
resources 'bugtracker' => 'https://github.com/michal-josef-spacek/Dicom-File-Detect/issues';
resources 'homepage' => 'https://github.com/michal-josef-spacek/Dicom-File-Detect';
resources 'repository' => 'git://github.com/michal-josef-spacek/Dicom-File-Detect';
test_requires 'English' => 0;
test_requires 'Error::Pure::Utils' => 0.22;
test_requires 'File::Object' => 0.05;
test_requires 'Test::More' => 0;
test_requires 'Test::NoWarnings' => 0;
tests_recursive;
version '0.06';

# Run.
WriteAll();
